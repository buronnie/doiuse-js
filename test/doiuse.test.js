const doiuse = require('../src/doiuse');

describe('doiuse', () => {
	it('should support filename as a string', async () => {
		const query = 'ie 11';
		const unsupportedFeatures = await doiuse(query, './test/data/fixture1.js');
		expect(unsupportedFeatures).toMatchSnapshot();
	});

	it('should return all unsupported features for the browsers whose usage > 1%', async () => {
		const query = '> 1%';
		const unsupportedFeatures = await doiuse(query, ['./test/data/fixture1.js']);
		expect(unsupportedFeatures).toMatchSnapshot();
	});
});
