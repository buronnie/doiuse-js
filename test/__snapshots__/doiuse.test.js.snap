// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doiuse should return all unsupported features for the browsers whose usage > 1% 1`] = `
Array [
  Object {
    "file": "./test/data/fixture1.js",
    "unsupported_features": Object {
      "arrow function": Array [
        "ie 11",
        "op_mini all",
      ],
      "class": Array [
        "ie 11",
        "op_mini all",
      ],
      "const": Array [
        "op_mini all",
      ],
      "export": Array [
        "and_uc 11.8",
        "ie 11",
        "op_mini all",
        "samsung 6.2",
      ],
      "generator": Array [
        "ie 11",
        "op_mini all",
      ],
      "import": Array [
        "and_uc 11.8",
        "ie 11",
        "op_mini all",
        "samsung 6.2",
      ],
      "template literal": Array [
        "ie 11",
        "op_mini all",
      ],
    },
  },
]
`;

exports[`doiuse should support filename as a string 1`] = `
Array [
  Object {
    "file": "./test/data/fixture1.js",
    "unsupported_features": Object {
      "arrow function": Array [
        "ie 11",
      ],
      "class": Array [
        "ie 11",
      ],
      "export": Array [
        "ie 11",
      ],
      "generator": Array [
        "ie 11",
      ],
      "import": Array [
        "ie 11",
      ],
      "template literal": Array [
        "ie 11",
      ],
    },
  },
]
`;
